extensions/amp-story-shopping/0.1/amp-story-shopping-tag.js

```diff
@@ -44,10 +48,32 @@ export class AmpStoryShoppingTag extends AMP.BaseElement {
 
     /** @param {!ShoppingConfigDataDef} tagData */
     this.tagData_ = null;
+
+    /** @param {?AmpElement} element */
+    this.shoppingAttachment_ = null;
+
+    /** @param {!AmpElement} element */
+    this.shoppingTagEl_ = null;
   }
 
   /** @override */
   buildCallback() {
+    /* This prevents the shopping tag component from building if there is no shopping attachment. */
+    const pageElement = closestAncestorElementBySelector(
+      this.element,
+      'amp-story-page'
+    );
+
+    const pageAttachment = childElementByTag(
```

Comment:
Nit: you can get rid of the `pageAttachment` variable and just set `this.shoppingAttachment_ = childElementByTag...`